Proposed charter changes from RFC (#23)
o3de/sig-security: governance/SIG Security Charter.md

# SIG Security Charter

## Overview of SIG

> SIG Security is a special interest group for O3DE code security that works as a central knowledge advisory group and  provides resources, consultation, reporting mechanisms  and information for security compliance, and security related issue resolution. This SIG also helps in keeping security patching up to date, and provide the basic services related to  APPSEC (Application Security) of O3DE.

Reviewer: nit: "is a special interest group for O3DE security" - remove code?

nit: "reporting mechanisms and" - looks like two spaces between "mechanisms" and "and"

Reviewer 2: I feel like this paragraph would be more readable if we simplified the sentence structure. What do you think about:

"SIG Security is a special interest group for O3DE security. It serves as an advisory group for security related issues like compliance, security issue resolution, and security patching. It's also the group responsible for maintaining vulnerability reporting mechanisms and assisting with other tasks that fall under the [Application Security (AppSec)](https://en.wikipedia.org/wiki/Application_security) umbrella of concern."

> SIG Security is a special interest group for O3DE code security that works as a central knowledge advisory group and  provides resources, consultation, reporting mechanisms  and information for security compliance, and security related issue resolution. This SIG also helps in keeping security patching up to date, and provide the basic services related to  APPSEC (Application Security) of O3DE.

Author: I can agree with that. @lmbr-pip @dshmz to commit in this and any other suggestions

## Goals

- To be set after rest of the charter is reviewed.

> ## Scope

Reviewer 2: As worded, I don't really understand the point of this section in addition to the below "In scope" section. It also reads more like a "Goals" section, esp. the 2 bullets re: Identifying potential risks in the code and defining best practices. Can we remove it, or is it part of some standard charter template?

Reviewer: Its part of the standard O3DE charter template.

> -  Involves gathering information about security flaws. 

Reviewer 3: To match stance of other Scope items:
`-  Gather information about...`

Author: ```suggestion
-  Gather information about security flaws. 
```

-  Identify potential security risks/gaps in Software.
-  Define best practices for ensuring secure code.

## In scope

- Triage and Acceptance of incoming security issues.

> - Creating mechanisms for intake of private security issues from any source.

Reviewer: "Creation and maintenance of mechanisms"...?

Reviewer: Added suggestion below.

> - Creating mechanisms for intake of private security issues from any source.

Reviewer: ```suggestion
- Creation and maintenance of mechanisms for the secure intake of vulnerability issues, including any security reporting email lists for O3DE.
```

- Management of GitHub security advisories for the O3DE repros.

> - Management of security reporting email and initial disclosure lists.

Reviewer: Maybe split and combine first part with first item and make "management of security disclosures" own item?

> - Management of security reporting email and initial disclosure lists.

Reviewer: ```suggestion
- Management of any security disclosure mechanisms including email lists.
```

> - Provides mechanisms for reporting security issues manually

Reviewer: This feels a repeat of first item.

> - Provides mechanisms for reporting security issues manually

Reviewer: ```suggestion
```

> - Provides tooling and automation for security issue identification and running campaigns to resolve identified security issues, including deprecation of software components.

Reviewer: Split into two.. "Automation of security issue detection"  and "running campagins across O3DE to resolve..."

> - Provides tooling and automation for security issue identification and running campaigns to resolve identified security issues, including deprecation of software components.

Reviewer: ```suggestion
- Assists in the selection or development of tooling and automation for security issue identification.
- Runs security campaigns to resolve identified security issues, including deprecation of software components.
```

- Provide SPDX (Software Package Data Exchange) scanning tools to detect violations.

## Cross-cutting Processes

> - Disseminating information to other sigs for action / update/ fixing

Reviewer: Maybe add something about driving dates and resources with impacted SIGs? SIG-Security does not own/provide resources for patching issues outside of code SIG-Security directly owns.

- Review and provide feedback on SIG presented features and implementation
- Provide SPDX Licensing report violations to corresponding SIGs

> - Advise SIGs on security level of new libraries 

Author: ```suggestion
- Advise SIGs on secure use of new libraries 
```

- Provide web and docs team with links to campaign information stored in SIG's repository
- Provide consultation for compliance automation scanning at periodic intervals and point release intervals for each potential issue and report violations to appropriate SIGs.
- Advise Technical steering committee and Governing board of potential threats and unresolved issues.
- Items that span or require other SIGs or groups and how it relates to this SIG’s responsibilities

## Out of Scope
- Not responsible for fixing any detected or reported security issues.
- Not responsible for enforcing or acting upon any legal compliance issues.
- Not responsible for implementing security or IP detection tools into automation or build chains, but may advise SIGs on where to implement.
- Items that are optional or are not the responsibility of this SIG.

> **SIG Links and lists:**

Reviewer: ```suggestion
## SIG Links and lists
```

- Joining this SIG - https://github.com/orgs/o3de/teams/sig-security
- Slack/Discord - https://discord.com/channels/805939474655346758/816043899477950475 
- Mailing list - https://lists.o3de.org/g/sig-security
- Issues/PRs - https://github.com/o3de/sig-security/issues/
- Meeting agenda & Notes - https://github.com/o3de/sig-security/issues/1 

> **Roles and Organization Management**

Reviewer: ```suggestion
## Roles and Organization Management
```

SIG Security adheres to the standards for roles and organization management as specified by <sig-governance>. This SIG opts in to updates and modifications to <sig-governance>

> **Individual Contributors**

Reviewer: ```suggestion
##I# ndividual Contributors

Should provide steps to reproduce, links and examples for security flaws, discoveries, or potential issues. Should raise potential security issues through the [appropriate mechanisms](https://github.com/o3de/sig-security/blob/main/rfcs/rfc-sec-20220215-1.md)
```

Additional information not found in the sig-governance related to contributors.

> **Maintainers**

Reviewer 4: Removing the expected role activities makes this weaker, but may be more accurate to the current state of the SIG guidance (I've been unable to attend meetings, and am unaware)

Reviewer: I think the intent is to only call things out where this SIG diverges from standard role behaviours. 

I would add that we expect maintainers to be able to evaluate security issues and ensure they are correctly reported with sufficient information to accept. However we can add maintainer expectations once we have issue response runbook in place.

Reviewer: Fixed below. Resolving.

> **Maintainers**

Reviewer: ```suggestion
###Maintainers

* Run issue triage when SIG-Chair(s) are unavailable. Must validate and verify that submitted issue or flaw exists and should be accepted by SIG.
* Help the SIG stay on top of any security concerns:
    * Engage with the [SIR team process](https://github.com/o3de/sig-security/blob/main/rfcs/rfc-sec-20220316-1.md)
    * (Optionally) Monitor GitHub issues for any security impacting issues and raise with SIG.
    * (Optionally) Monitor DIscord for any security impacting issues and raise with SIG.

```

Additional information not found in the sig-governance related to contributors

> **Additional responsibilities of Chairs**

Reviewer: ```suggestion
### Additional responsibilities of Chairs
```

Additional information not found in the sig-governance related to SIG Chairs

> **Subproject Creation**

Reviewer: ```suggestion
## Subproject Creation
```

Additional information not found in the sig-governance related to subproject creation

> **Deviations from sig-governance**

Reviewer: ```suggestion
## Deviations from sig-governance
```
